### DISCUSSION TITLE
[Q/Help] mini.completion: weird markdown render in signature help vs lsp.signature_help

### DISCUSSION THREAD

**Discussion by ghost** [nvim-mini/mini.nvim#1426] [Q&A] [answered]
Hi, 

first I want to thank you for your awesome mini library. I've been using some of your mini plugins for some time now, including mini.completion.

I specially like the "one-line" signature help, because on other plugins I tried, this kind of feature mostly gives you the full blown help description as well which takes away real estate. But in my workflow at least, I have to know/look-up a function anyway before using it and all I need is the reminder about arguments, their type and order.

---

There is one thing that has been bugging me for a while and I hope you could help me with what I'm doing wrong.

Basically the mini.completion's signature help is rendering some special characters weirdly. E.g. below (first image) the closing parenthesis and the underscore are highlighted in red when compared to `vim.lsp.buf.signature_help` (second image). Also the markdown characters like ` `` ` (backticks) and `*` are hidden in `vim.lsp.buf.signature_help` and `vim.lsp.buf.hover` (third image) while they are visible in mini.completion.

This is my nvim config: https://github.com/myzb/dotfiles/tree/master/.config/nvim

mini.completion signature help:
![Screenshot From 2024-12-22 12-57-26](https://github.com/user-attachments/assets/4dd1e133-23e5-4fc3-ab39-d19cc29f8ce7)

vim.lsp.buf.signature_help:
![image](https://github.com/user-attachments/assets/9994e4f6-a83d-4055-9128-ba3e1b50ebf3)

vim.lsp.buf.hover:
![image](https://github.com/user-attachments/assets/be6e573d-b592-4f9b-92c8-768cf04f7cc4)

Could you help me finding what I'm doing wrong?

Thank you a lot!
myzb

**Comment by echasnovski** [comment, marked as answer]
@myzb, thanks for kind words!

Let's first unpack something here for 'mini.completion': the small usually one-line window is indeed signature help window, while the big one beside completion item is info window.

Second, you are doing nothing wrong here. The markdown-like symbols is just how 'mini.completion' displays both windows: it doesn't conceal them. I think the motivation was that with them hidden it might sometimes be too wide (with extra trailing whitespace) which throws sometimes off.

I am not sure about the red highlighting. I vaguely remember figuring this out some time in the past, but don't remember.

The difference between 'mini.completion' and `vim.lsp.buf` methods is simply because the first was not significantly updated in years. Luckily, I plan to divert my attention to it for a bit after 'mini.snippets' is merged. I've mentioned these issues in #1020 so as to not forget about them.

**Reply by ghost** [reply]
Thanks for the quick reply, I'll be eagerly awaiting for mini.snippets and what comes next for mini.completion. 

Regarding the red highlighting it appears to be some error highlight. On "gbprod/nord.nvim" (my current theme) it is more obvious because the highlight has a red background. On "AlexvZyl/nordic.nvim" on the other hand, it is less obvious, because the theme uses a redish foreground color for that error highlight.

**Comment by echasnovski** [comment]
For completeness, these issues appear to be solved after resolving #1020:
- Special characters are concealed.
- The red background highlighting indeed comes from `Error` highlight group and indicates the error for syntax parsing. I believe the current approach to highlighting should not have that.
